### DISCUSSION TITLE
Physical Media dumping/bridging/launching

### DISCUSSION THREAD

**Discussion by sairuk** [retronas/discussions#75] [General]
There was a [thread](https://twitter.com/RonaldSDean/status/1513591951790067724) on twitter about this recently. While we had floppy, tape and hdd ideas but I'd say we'd overlooked carts _apart_ from it rattling around in the back of my mind to do some kind of webman for MiSTer oneday.

I personally no longer store sets of games (MAME only these days), I instead dump my real media so I can verify it against scene data.

So this is intended as a more generalised discussion about getting your physical media onto RetronNAS

So have at it, we already have uCON64 available for install and we want to look at MPF (Redumps disc dumping tools).

Suggestions are open.

**Comment by sairuk** [comment]
We would like someone with a Retrode to plug it into a linux box and run `lsusb` then give us the output so we can see if this is some support we can add.

**Comment by jamescadd** [comment]
A system that worked well with the redump tools and knew the right place to store games would be nice.  I've recently been ripping PS2 games with the redump tools, then copying the ISOs to the right folder (CD or DVD) on RetroNAS via Samba share to Windows.  I don't rip any carts though so maybe this is off topic.

Ideally I would like RetroNAS to have a webpage that looks like the Plex library and has a button to rip disc (or I guess cart).  Bonus if there is also an option to import game sets for those that use them.  I know there are a bunch of library management tools out there and wonder if there's something that can be pulled in to RetroNAS that's already good enough.

**Comment by danmons** [comment]
Optical media ripping of pre-copy-protection systems like PS1, PS2, MegaCD, PCE-CD, etc would all definitely be doable.  Any standard CD or DVD ROM drive, including USB connected, can be used to "ddrescue" media over to an ISO image, initiated by a web interface.

This has been on my want list for a while.  Likewise simple floppy images (MS-DOS floppies or other standard 1.44MB formatted disks).   These will likely be added first to test the GUI side of things while keeping the dumping part rather trivial.  

Ditto for pushing back out to devices like Gotek/Flashfloppy, or creating IMG/VHD type images of a given size and disk format. 

More complex stuff includes anything with copy protection which could require CLI  version of the Sabretools Media Preservation Frontend stuff (Sega Saturn comes to mind), support for floppy flux readers like GreaseWeazle for non-PC hardware floppies (Amiga, etc), and of course cartridge dumpers (uCON64, etc), all of which need some complex arguments, specific hardware devices and ports,  and system knowledge.  Huge job for those.

**Comment by shw-dev** [comment]
I have a Retrode, but the output of lsusb doesn't seem to be very informative. It just outputs the bus and device IDs, is there anything specific I should be looking for?

**Reply by sairuk** [reply]
anything to identify and/or target the device, we are in discovery phase, looking for a name, product/vendor ids etc for now

There is more info is available with `-v` switch so `lsusb -v` 

alternatively perhaps `udevadm info -a <dev path>` may get us the info we need, see how you go

**Reply by shw-dev** [reply]
```
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               1.10
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0         8
  idVendor           0x0403 Future Technology Devices International, Ltd
  idProduct          0x97c1
  bcdDevice           26.00
  iManufacturer           1
  iProduct                2
  iSerial                 0
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           82
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass         8 Mass Storage
      bInterfaceSubClass      6 SCSI
      bInterfaceProtocol     80 Bulk-Only
      iInterface              0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 No Subclass
      bInterfaceProtocol      0 None
      iInterface              0
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     176
         Report Descriptors:
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval              10
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      50
         Report Descriptors:
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x86  EP 6 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval              10
```

**Reply by sairuk** [reply]
perfect thanks, be a good reference for when we can get to this.

```
  idVendor           0x0403 Future Technology Devices International, Ltd
  idProduct          0x97c1
```

**Reply by shw-dev** [reply]
Great! Let me know if you need anything else

**Comment by sairuk** [comment]
DiscImageCreator (redump.org) is now available in #RetroNAS under the experimental menu. 

I do not have a compatible USB drive to test these tools so if you do and can test dumping on an rpi (or whatever) and experience with DIC let us know how you go

It will be considered an experimental addition for a while yet

For usage information see: http://wiki.redump.org/index.php?title=Dumping_Guides

**Reply by sairuk** [reply]
This is also experimental because I had to fork one of the dependents (DVDAuth) and patch it to compile on arm, but i'm winging it there so it may not function
